### PR TITLE
Support a “modal” prop for tabbed iOS apps

### PR DESCRIPTION
**Pull request by kevboh**
This adds support for a `modal` boolean prop when starting a tab-based app on iOS. If set to `true`, tapping that tab will not select it and instead emit a `"modalTabSelected"` event navigation event, which can be listened to in order to present a modal on tab tap (or do anything else, I suppose).

If there's a better/accepted way to do this already, please let me know and close this PR. Thanks.

### CODE REVIEW

**Comment by gvillenave** [comment]
+1, can we get this merged?

**Comment by adnanbabar** [comment]
@kevboh Thank you so much for this!

I merged your pull into my Fork and it's working like a dream!

Hope your pull gets merged soon.

File: docs/top-level-api.md [hunk 1]
```diff
@@ -44,7 +44,8 @@ Navigation.startTabBasedApp({
       title: 'Screen One', // title of the screen as appears in the nav bar (optional)
       titleImage: require('../img/titleImage.png'), // iOS only. navigation bar title image instead of the title text of the pushed screen (optional)
       navigatorStyle: {}, // override the navigator style for the tab screen, see "Styling the navigator" below (optional),
-      navigatorButtons: {} // override the nav buttons for the tab screen, see "Adding buttons to the navigator" below (optional)
+      navigatorButtons: {}, // override the nav buttons for the tab screen, see "Adding buttons to the navigator" below (optional)
+      modal: false // prevent tab selection and send a 'modalTabSelected' navigator event instead (optional)
```
Comment (wbyoung) [inline docs/top-level-api.md]: This should contain `iOS only` for clarity.

**Comment by adnanbabar** [comment]
@guyca Any update on this, please??

**Comment by sunb123** [comment]
Can this be done in RNN V2?

**Comment by guyca** [comment]
@sunb123 This will be supported in v2, tracking in #2766

**Comment by brenwell** [comment]
Is there an example usage anywhere?

**Comment by brenwell** [comment]
Was this reverted?

**Comment by quantosapplications** [comment]
Hello, is there a way this will be merged and will work on android too?

**Pull request merged**